sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient_helpers.cs

```diff
@@ -42,18 +43,24 @@ public virtual Uri GetClientAccessUri(string userId = default, string[] roles =
                 claims.Add(role);
             }
 
-            string endpoint = _endpoint.AbsoluteUri;
+            string endpoint = this.endpoint.AbsoluteUri;
             if (!endpoint.EndsWith("/", StringComparison.Ordinal))
             {
                 endpoint += "/";
             }
-            var audience = $"{endpoint}client/hubs/{_hub}";
+            var audience = $"{endpoint}client/hubs/{hub}";
 
-            string token = WebPubSubAuthenticationPolicy.GenerateAccessToken(audience, claims, _credential, expireAfter);
+            // this is a hack. The generator needs to store the credential in a field.
+            if (credential == null)
+            {
+                HttpPipeline pipeline = Pipeline;
```

Comment:
This is pretty explicit, but you can consider using a discard `_` to be even more so.

Reply:
I removed all generated ctors and this is not needed anymore. Removed.